cep-sigstore-predicate.md

```diff
@@ -12,61 +12,182 @@
+  Another key misuse-resistance contribution within Sigstore is *machine
+  identities*. A machine identity behaves similarly to a human identity
+  (Alice or Bob), but identifies a machine instead of a human. For example,
+  `github.com/example/example/.github/workflows/release.yml@refs/tags/v1.2.3`
+  could be the machine identity of a GitHub Actions workflow that ran from
+  `release.yml` within `example/example` against the `v1.2.3` tag.
+
+## Motivation
+
+The conda ecosystem contains metadata that answers the following questions,
+in part or in full:
+
+* _Who_ (or _what_) published this package?
+* _What_ is the package's hash?
+* _Where_ was this package _published from_, and where _to_?
+* _When_ was this package published?
+
+However, this metadata is not currently **cryptographically verifiable**:
+the consuming party must either trust it as presented, or verify it manually
+against independent sources of truth (such as a project's release history).
+
+Attestations that present this metadata in a cryptographically
+verifiable manner are desirable for a number of reasons:
+
+* Package maintainers wish to demonstrate the integrity and authenticity
+  of their package uploads;
+* Individual downstream users wish to verify the integrity and authenticity of
+  packages they consume, without placing additional trust in the
+  channel or channel's hosting server;
+* Attestations change the sophistication and risk profile for attackers in
+  defenders' favor: the attacker must be sufficiently sophisticated
+  to access private key material, *and* have a risk tolerance profile that
+  accepts exposure via auditable transparency logs.
+
+More broadly, attestation schemes like the one proposed in this CEP have
+seen adoption in similar and related ecosystems:
+
+* Python (PyPA/PyPI): [PEP 740] and [PyPI - Attestations]
+* NodeJS (npm): [npm - Generating provenance statements]
+* Ruby (RubyGems): [rubygems/release-gem]
 
-### Sigstore Attestations
-
-Sigstore attestations are cryptographic statements about software artifacts that provide:
-
-- Authenticity: Proof of who created/signed the artifact
-- Integrity: Verification that the artifact hasn't been tampered with
-- Transparency: Public record of signatures in a tamper-evident log
-
-### Key Components
+## Specification
 
-- Predicates: JSON documents containing metadata about the signing event, using the `in-toto` format
-- Signatures: Cryptographic proofs made using ephemeral keys
-- Rekor: A tamper-evident log that stores attestations
-- Fulcio: A certificate authority that issues short-lived certificates
+### Attestation format
 
-In this document, we want to standardize the sigstore predicate for conda packages. The bundle format to be used for sigstore attestations is the `v0.3` bundle format.
+This CEP proposes the following attestation statement layout, using the
+[in-toto Statement schema]:
 
-## Specification
+- `predicateType` **MUST** be `https://schemas.conda.org/publish-v1.json`
```

Comment:
Flagging: based on some of the feedback in the upstream PR, I renamed this to `publish-v1.json`, since that (1) describes the attestation's semantics and (2) is broadly consistent with what PyPI standardized for its attestation.

Per https://github.com/conda/ceps/pull/112#discussion_r1990055851 another good option here would be `https://schemas.conda.org/attestations/publish/v1`.

Reply:
totally fine with me!